### PR TITLE
Add mineclone2, mineclonia and mineclone5 support

### PR DESCRIPTION
**Pull request by JamesClarke7283**
This PR adds support for mineclone2 and related subgames.

It currently does not support the mg mod, but i will work to fix this.

### CODE REVIEW

File: _config.txt [hunk 1]
```diff
@@ -78,3 +78,23 @@ moreores.mithril_max_depth = -2048          -- For v6 mapgen, -256 fits better
...
+moreores.silver_min_depth_high = -10
+moreores.silver_max_depth = -11
+moreores.silver_min_depth = -57
+
+moreores.mithril_max_depth_high = 0
+moreores.mithril_min_depth_high = -20
+moreores.mithril_max_depth = -21
+moreores.mithril_min_depth = -57
+
+end
```
Comment (Calinou) [inline _config.txt]:
```suggestion
if minetest.get_modpath("mcl_core") then
	-- Example adjustments for MineClone2
	moreores.tin_max_depth_high = 0
	moreores.tin_min_depth_high = -10
	moreores.tin_max_depth = -11
	moreores.tin_min_depth = -57
	
	-- Similar adjustments for silver and mithril
	moreores.silver_max_depth_high = 0
	moreores.silver_min_depth_high = -10
	moreores.silver_max_depth = -11
	moreores.silver_min_depth = -57
	
	moreores.mithril_max_depth_high = 0
	moreores.mithril_min_depth_high = -20
	moreores.mithril_max_depth = -21
	moreores.mithril_min_depth = -57
end
```

File: init.lua [hunk 2]
```diff
@@ -25,11 +25,28 @@ end
...
+default_stone_sounds = default.node_sound_stone_defaults()
+default_metal_sounds = default.node_sound_metal_defaults()
+end
+
 
 -- Returns the crafting recipe table for a given material and item.
 local function get_recipe(material, item)
+	if is_mcl_core_present then
+        material = material:gsub("default:", "mcl_core:")
+    end
```
Comment (Calinou) [inline init.lua]:
```suggestion
	if is_mcl_core_present then
		material = material:gsub("default:", "mcl_core:")
	end
```

File: init.lua [hunk 3]
```diff
@@ -75,6 +92,11 @@ local function get_recipe(material, item)
 end
 
 local function add_ore(modname, description, mineral_name, oredef)
+
+	if mineral_name == "copper" and is_mcl_copper_present then
+        return
+    end
```
Comment (Calinou) [inline init.lua]:
```suggestion
		return
	end
```

File: mod.conf [hunk 4]
```diff
@@ -1,5 +1,5 @@
 name = moreores
 description = Adds new ore types.
-depends = default
-optional_depends = carts,farming,frame,mg,toolranks
+optional_depends = carts,farming,frame,mg,toolranks,mcl_core,mcl_sounds
 min_minetest_version = 5.0.0
+supported_games mineclone2,mineclonia,minetest_game
```
Comment (Calinou) [inline mod.conf]:
```suggestion
supported_games = mineclone2,mineclonia,minetest_game
```

File: init.lua [hunk 5]
```diff
@@ -25,11 +25,29 @@ end
 -- `frame` support
 local use_frame = minetest.get_modpath("frame")
 
-local default_stone_sounds = default.node_sound_stone_defaults()
-local default_metal_sounds = default.node_sound_metal_defaults()
+local is_mcl_core_present = minetest.get_modpath("mcl_core") ~= nil
+local is_mcl_sounds_present = minetest.get_modpath("mcl_sounds") ~= nil
+local is_mcl_copper_present = minetest.registered_items["mcl_copper:copper_ingot"] ~= nil
+local stone_ingrediant = mcl_core_modpath and "mcl_core:stone" or "default:stone"
+local copper_ingrediant = mcl_core_modpath and "mcl_copper:copper_ingot" or 'default:copper_ingot'
```
Comment (Calinou) [inline init.lua]: `ingredient` should have a `e`, not `a`. Please update all references to these variables as well.

**Comment by JamesClarke7283** [comment]
It should be good to go now @Calinou

**Comment by Calinou** [comment]
CI fails due to Luacheck errors:

```
Luacheck.................................................................Failed
- hook id: luacheck
- exit code: 1

Checking init.lua                                 4 warnings

    init.lua:36:7: value assigned to variable default_stone_sounds is unused
    init.lua:37:7: value assigned to variable default_metal_sounds is unused
    init.lua:40:24: accessing undefined variable mcl_sounds
    init.lua:41:24: accessing undefined variable mcl_sounds

Checking mg.lua                                   OK

Total: 4 warnings / 0 errors in 2 files
```

**Comment by JamesClarke7283** [comment]
> CI fails due to Luacheck errors:
> 
> ```
> Luacheck.................................................................Failed
> - hook id: luacheck
> - exit code: 1
> 
> Checking init.lua                                 4 warnings
> 
>     init.lua:36:7: value assigned to variable default_stone_sounds is unused
>     init.lua:37:7: value assigned to variable default_metal_sounds is unused
>     init.lua:40:24: accessing undefined variable mcl_sounds
>     init.lua:41:24: accessing undefined variable mcl_sounds
> 
> Checking mg.lua                                   OK
> 
> Total: 4 warnings / 0 errors in 2 files
> ```

Fixed

**Review by Calinou — APPROVED** [review]
Thanks!

**Pull request merged**